persistence/sql/sql_process_store_common.go

```diff
@@ -32,7 +31,15 @@ func insertAsyncStateExecution(
 	stateInput []byte,
 	stateInfo []byte,
 ) error {
-	commandResultsBytes, err := persistence.FromCommandResultsToBytes(xdbapi.CommandResults{})
+	commandResultsBytes, err := persistence.FromCommandResultsJsonToBytes(persistence.NewCommandResultsJson())
+	if err != nil {
+		return err
+	}
+
+	// set default value for commandRequest to prepare for the wait_until api call
+	commandRequestBytes, err := persistence.FromCommandRequestToBytes(xdbapi.CommandRequest{
```

Comment:
The reason I put a default one here is that, if using empty struct, when getting the value back from db, the decoding part seems throwing error. You can try it on your local machine to verify. Or do you have a better way to avoid the error throwing?

Reply:
It's related to this comment: https://github.com/xdblab/xdb/pull/73#discussion_r1378733126